src/main/webapp/app/shared/virtual-scroll/virtual-scroll.component.ts

```diff
@@ -0,0 +1,255 @@
+/**
+ * Component based on Open Source Project ag-virtual-scroll
+ * https://github.com/ericferreira1992/ag-virtual-scroll
+ *
+ */
+
+import { Component, ElementRef, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, Renderer2, SimpleChanges, ViewChild } from '@angular/core';
+import { NavigationStart, Router } from '@angular/router';
+import { VirtualScrollRenderEvent } from 'app/shared/virtual-scroll/virtual-scroll-render-event.class';
+import { ITEMS_PER_PAGE } from 'app/shared/constants/pagination.constants';
+
+@Component({
+    selector: 'virtual-scroll',
+    templateUrl: './virtual-scroll.component.html',
+    styleUrls: ['virtual-scroll.style.css'],
+})
+export class VirtualScrollComponent implements OnInit, OnChanges, OnDestroy {
+    @ViewChild('itemsContainer', { static: true }) private itemsContainerElRef: ElementRef<HTMLElement>;
+
+    @Input('items') public originalItems: any[] | undefined = [];
+    @Input() forceReload: boolean;
+    @Output() forceReloadChange = new EventEmitter<boolean>();
+    @Output() private onItemsRender = new EventEmitter<VirtualScrollRenderEvent<any>>();
+
+    minRowHeight = 126.7;
+
+    public prevOriginalItems: any[] = [];
+    public domTreeItems: any[] = [];
+
+    public currentScroll = 0;
+    public contentHeight = 0;
+    public paddingTop = 0;
+
+    public startIndex = 0;
+    public endIndex = 0;
+
+    private scrollIsUp = false;
+    private lastScrollIsUp = false;
+
+    previousItemsHeight: any[] = [];
+
+    scrollListener: any;
+    focusInListener: any;
+
+    constructor(private elRef: ElementRef<HTMLElement>, private renderer: Renderer2, private router: Router) {}
```

Comment:
```suggestion
    constructor(private elementRef: ElementRef<HTMLElement>, private renderer: Renderer2, private router: Router) {}
```

I suggest to refactor this in IntelliJ